### ISSUE TITLE
[JS] [Dotprompt] Optional Schema Fields create "Proto field is not repeating, cannot start list" on Gemini API

### ISSUE THREAD

**Issue by i2amsam** [firebase/genkit#1393]
**Describe the bug**
Including optional schema fields in dotprompts results in "Invalid JSON payload received. Unknown name "type" at 'generation_config.response_schema.properties[2].value.items.properties[1].value': Proto field is not repeating, cannot start list" from Gemini API

myprompt.prompt
```
---
model: googleai/gemini-1.5-flash
input:
  schema:
    theme?: string
  default:
    theme: "pirate"
output:
  schema:
    title: string # string, number, and boolean types are defined like this
    subtitle?: string # optional fields are marked with a `?`
---
Invent a menu item for a {{theme}} themed restaurant.
```

```
const myPrompt = ai.prompt('myprompt');

(async () => {
    console.log(((await myPrompt({theme: "foodie"})).output));
})();
```

```
GoogleGenerativeAIFetchError: [GoogleGenerativeAI Error]: Error fetching from https://generativelanguage.googleapis.com/v1beta/models/gemini-1.5-flash:generateContent: [400 Bad Request] Invalid JSON payload received. Unknown name "type" at 'generation_config.response_schema.properties[2].value.items.properties[1].value': Proto field is not repeating, cannot start list. [{"@type":"type.googleapis.com/google.rpc.BadRequest","fieldViolations":[{"field":"generation_config.response_schema.properties[2].value.items.properties[1].value","description":"Invalid JSON payload received. Unknown name \"type\" at 'generation_config.response_schema.properties[2].value.items.properties[1].value': Proto field is not repeating, cannot start list."}]}]
    at handleResponseNotOk (/home/user/mytestapp/node_modules/@google/generative-ai/dist/index.js:414:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async makeRequest (/home/user/mytestapp/node_modules/@google/generative-ai/dist/index.js:387:9)
    at async generateContent (/home/user/mytestapp/node_modules/@google/generative-ai/dist/index.js:832:22)
    at async ChatSession.sendMessage (/home/user/mytestapp/node_modules/@google/generative-ai/dist/index.js:1146:9) {
  status: 400,
  ```
  
  Removing the optional subtitle field allows the generation to work as expected.

**Comment by pavelgj** [comment]
this should be fixed in https://github.com/firebase/genkit/pull/1411

**Comment by i2amsam** [comment]
validated fix in 0.9.5 👍 💯
